src/LiveComponent/assets/src/live_controller.js

```diff
@@ -0,0 +1,635 @@
+        // However, there is an edge case where the user changes data-model="post"
+        // and then, for some reason, they don't want an immediate re-render.
+        // Then, then modify the data-model="post.title" field. In theory,
+        // we should be smart enough to convert the post data - which is now
+        // the string "4" - back into an array with [id=4, title=new_title].
+        this.dataValue = setDeepData(this.dataValue, modelName, value);
+
+        directive.modifiers.forEach((modifier => {
+            switch (modifier.name) {
+                // there are currently no data-model modifiers
+                default:
+                    throw new Error(`Unknown modifier ${modifier.name} used in data-model="${model}"`)
+            }
+        }));
+
+        if (shouldRender) {
+            // clear any pending renders
+            this._clearWaitingDebouncedRenders();
+
+            // todo - make timeout configurable with a value
+            this.renderDebounceTimeout = setTimeout(() => {
+                this.renderDebounceTimeout = null;
+                this.$render();
+            }, this.debounceValue || DEFAULT_DEBOUNCE);
+        }
+    }
+
+    _makeRequest(action) {
+        let [url, queryString] = this.urlValue.split('?');
+        const params = new URLSearchParams(queryString || '');
+
+        const fetchOptions = {
+            headers: {
+                'Accept': 'application/vnd.live-component+json',
+            },
+        };
+
+        if (action) {
+            url += `/${encodeURIComponent(action)}`;
+
+            if (this.csrfValue) {
+                fetchOptions.headers['X-CSRF-TOKEN'] = this.csrfValue;
+            }
+        }
+
+        if (!action && this._willDataFitInUrl()) {
+            buildSearchParams(params, this.dataValue);
+            fetchOptions.method = 'GET';
+        } else {
+            fetchOptions.method = 'POST';
+            fetchOptions.body = buildFormData(this.dataValue);
+        }
+
+        // todo: make this work for specific actions, or models
+        this._onLoadingStart();
+        const paramsString = params.toString();
+        const thisPromise = fetch(`${url}${paramsString.length > 0 ? `?${paramsString}` : ''}`, fetchOptions);
+        this.renderPromiseStack.addPromise(thisPromise);
+        thisPromise.then((response) => {
+            // if another re-render is scheduled, do not "run it over"
+            // todo: think if this should behave differently for actions
+            if (this.renderDebounceTimeout) {
+                return;
+            }
+
+            const isMostRecent = this.renderPromiseStack.removePromise(thisPromise);
+            if (isMostRecent) {
+                response.json().then((data) => {
+                    this._processRerender(data)
+                });
+            }
+        })
+    }
+
+    /**
+     * Processes the response from an AJAX call and uses it to re-render.
+     *
+     * @todo Make this truly private
```

Comment:
There are a lot of TODOs here. Are all of them still relevant? Should some be fixed or write issues on?